x-pack/plugins/upgrade_assistant/__jest__/client_integration/kibana_deprecations/deprecations_list.test.ts

```diff
@@ -0,0 +1,138 @@
+import { KibanaTestBed, setupKibanaPage } from './kibana_deprecations.helpers';
+import { kibanaDeprecationsMockResponse } from './mocked_responses';
+
+const criticalDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.level === 'critical'
+);
+const warningDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.level === 'warning'
+);
+const configDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.deprecationType === 'config'
+);
+
+describe('Kibana deprecations table', () => {
+  let testBed: KibanaTestBed;
+  const { server } = setupEnvironment();
+  const deprecationService = deprecationsServiceMock.createStartContract();
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  beforeEach(async () => {
+    await act(async () => {
+      deprecationService.getAllDeprecations = jest
+        .fn()
+        .mockReturnValue(kibanaDeprecationsMockResponse);
+
+      testBed = await setupKibanaPage({
+        services: {
+          core: {
+            deprecations: deprecationService,
+          },
+        },
+      });
+    });
+
+    testBed.component.update();
+  });
+
+  test('renders deprecations', () => {
+    const { exists, table } = testBed;
+
+    expect(exists('kibanaDeprecations')).toBe(true);
+
+    const { tableCellsValues } = table.getMetaData('kibanaDeprecationsTable');
+
+    expect(tableCellsValues.length).toEqual(kibanaDeprecationsMockResponse.length);
+  });
+
+  it('refreshes deprecation data', async () => {
+    const { actions } = testBed;
+
+    await actions.table.clickRefreshButton();
+
+    expect(deprecationService.getAllDeprecations).toHaveBeenCalledTimes(2);
+  });
+
+  it('shows critical and warning deprecations count', () => {
+    const { find } = testBed;
+
+    expect(find('criticalDeprecationsCount').text()).toContain(criticalDeprecations.length);
+    expect(find('warningDeprecationsCount').text()).toContain(warningDeprecations.length);
+  });
+
+  describe('Search bar', () => {
+    it('filters by "critical" status', async () => {
+      const { actions, table } = testBed;
+
+      await actions.searchBar.clickCriticalFilterButton();
+
+      const { rows: criticalRows } = table.getMetaData('kibanaDeprecationsTable');
+
+      expect(criticalRows.length).toEqual(criticalDeprecations.length);
+
+      await actions.searchBar.clickCriticalFilterButton();
+
+      const { rows: allRows } = table.getMetaData('kibanaDeprecationsTable');
+
+      expect(allRows.length).toEqual(kibanaDeprecationsMockResponse.length);
+    });
+
+    it('filters by type', async () => {
+      const { component, table, actions } = testBed;
+
+      await actions.searchBar.clickTypeFilterDropdownAt(0);
+
+      // We need to read the document "body" as the filter dropdown options are added there and not inside
+      // the component DOM tree.
+      const configTypeFilterButton: HTMLButtonElement | null = document.body.querySelector(
```

Comment:
It might be helpful to explain in the comment that we expect the first item in the list to be the "Config" option.